designer/client/conditions/inline-conditions-definition.js

```diff
@@ -0,0 +1,192 @@
+import React from 'react'
+import { Condition, Field, Value } from './inline-condition-model'
+import { clone } from '../helpers'
+
+const conditionalOperators = {
+  'default': {
+    is: {},
+    'is not': {}
+  },
+  'TextField': {
+    matches: {}
+  },
+
+  getConditionals: (fieldType) => {
+    return Object.assign(conditionalOperators[fieldType] || {}, conditionalOperators['default'])
+  }
+}
+
+class InlineConditionsDefinition extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      condition: clone(props.condition) || {}
+    }
+  }
+
+  componentDidUpdate (prevProps, prevState, snapshot) {
+    if (this.props.expectsCoordinator !== prevProps.expectsCoordinator ||
+        this.props.fields !== prevProps.fields) {
+      const { condition } = this.state
+      const newCondition = this.props.fields[condition?.field?.name] ? this.state.condition : {}
+      this.setState({
+        condition: newCondition
+      })
+    }
+  }
+
+  onChangeCoordinator = e => {
+    const input = e.target
+
+    if (input.value && input.value.trim() !== '') {
+      const copy = clone(this.state.condition??{})
+      copy.coordinator = input.value
+      this.setState({
+        condition: copy
+      })
+    } else {
+      this.setState({
+        condition: {}
+      })
```

Comment:
can we do just one setState? `let update = {}`, set update inside the if block, then `this.setState(...condition: update)`. that way we can get rid of else

Reply:
ok